### PR TITLE
[JENKINS-56632] Jenkinsfile Runner CI Pipeline is unstable

### PR DESCRIPTION
**Pull request by fcojfernandez**
See [JENKINS-56632](https://issues.jenkins-ci.org/browse/JENKINS-56632)

This PR is a follow up for #99 and relies on it.

Jenkinsfile uses agents with the `docker` label. These agents are permanent agents which do not automatically cleanup workspaces or orchestrate parallel runs. If multiple runs are executed in parallel, it may lead to various workspace collisions. This PR try to clean up the workspace so the parallel executions succeed.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-ARG JENKINS_VERSION=2.150.3
+ARG JENKINS_VERSION=2.164.1
```
Comment (oleg-nenashev) [inline Dockerfile]: unrelated but LGTM

**Review by oleg-nenashev — COMMENTED** [review]
There is no workspace cleanup in the current implementation. Sinc ewe do not have access to ci.jenkins.io workspaces anyway, I suggest adding workspace cleanup using try/finally

**Comment by fcojfernandez** [comment]
@oleg-nenashev I tried using `try/finally` and `cleanWs` but ci.jenkins.io has not _[Workspace Cleanup Plugin](https://plugins.jenkins.io/ws-cleanup)_ as installed. See [commit 3d7bf05](https://github.com/jenkinsci/jenkinsfile-runner/pull/101/commits/3d7bf05590bd88d6377e4667d6df5d58bab924c6) and [build #5](https://ci.jenkins.io/blue/organizations/jenkins/Tools%2Fjenkinsfile-runner/detail/PR-101/5/pipeline/) where you can see the `java.lang.NoSuchMethodError: No such DSL method 'cleanWs' found among steps` error.

I also tried with `dir` and `deleteDir` (See [commit 600917c](https://github.com/jenkinsci/jenkinsfile-runner/pull/101/commits/600917cfba0e676373230978ec4f1b46b57d7361)), but since the tests execute the docker images as root, the files generated in volumes that are mounted are created as root, so the `deleteDir` execution fails: https://ci.jenkins.io/blue/organizations/jenkins/Tools%2Fjenkinsfile-runner/detail/PR-101/1/pipeline/

**Review by varyvol — APPROVED** [review]

**Comment by fcojfernandez** [comment]
@oleg-nenashev I'm merging this PR. Since the _Workspace Cleanup Plugin_ is not installed and some files are created as root, I cannot clean up the workspace. With this PR at least we prevent the collisions. If you know how to improve the clean-up, I'm totally happy to create a follow-up PR

**Pull request merged**